src/options/views/tab-settings/vm-editor.vue

```diff
@@ -37,6 +42,47 @@ export default {
       const toggled = selection === 'false' && 'true' || selection === 'true' && 'false';
       if (toggled) document.execCommand('insertText', false, toggled);
     },
+    async toggleStateHint() {
+      if (this.hint) {
+        this.hint = null;
+        return;
+      }
+      const HIDE_OPTS = [
+        // we activate only one mode: js
+        'mode',
+        // duh
+        'value',
+        // these accept only a function
+        'configureMouse',
+        'lineNumberFormatter',
+        'specialCharPlaceholder',
+      ];
+      const opts = {};
+      Object.entries({
+        ...(await import('codemirror')).defaults,
+        ...(await import('#/common/ui/code')).cmOptions,
+        ...options.get('editor'),
+      })
+      .sort(([a], [b]) => (a < b ? -1 : a > b))
+      .forEach(([k, v]) => {
+        if (!HIDE_OPTS.includes(k) && typeof v !== 'function') {
+          opts[k] = v;
+        }
+      });
```

Comment:
you seem to prefer a more verbose name like key/value, is it intentional and should I use it here too?

Reply:
It does not matter for such a simple case. I prefer `filter |> reduce` though.